crates/runtime/src/cow.rs

```diff
@@ -566,17 +566,18 @@ impl Drop for MemoryImageSlot {
         // over by the next MemoryImageSlot later.
         //
         // Since we're in drop(), we can't sanely return an error if
-        // this mmap fails. Let's ignore the failure if so; the next
-        // MemoryImageSlot to be created for this slot will try to overwrite
-        // the existing stale mappings, and return a failure properly
-        // if we still cannot map new memory.
+        // this mmap fails. Instead though the result is unwrapped here to
+        // trigger a panic if something goes wrong. Otherwise if this
+        // reset-the-mapping fails then on reuse it might be possible, depending
+        // on precisely where errors happened, that stale memory could get
+        // leaked through.
         //
         // The exception to all of this is if the `unmap_on_drop` flag
```

Comment:
pre-existing but this is `unmap_on_drop` while a few lines below we say `clear_on_drop` -- would you mind updating the comment while we're here?